### PR TITLE
"sapphirerapids": support for GCC/Clang/Intel/Oneapi

### LINKED ISSUE #75: Add for sapphire rapids
It would be nice if archspec had support for the sapphire rapids chipset.

### PR DESCRIPTION
**Pull request by stephenmsachs**
fixes #75

### CODE REVIEW

File: cpu/microarchitectures.json [hunk 1]
```diff
@@ -1433,6 +1433,92 @@
...
+            "versions": "2021.2:",
+            "flags": "-march={name} -mtune={name}"
+          }
+        ],
+        "dpcpp": [
+          {
+              "versions": "2021.2:",
+            "flags": "-march={name} -mtune={name}"
+          }
+        ]
```
Comment (alalazo) [inline cpu/microarchitectures.json:1519]: @rscohn2 Could you please double check these ?

Comment (rscohn2) [reply cpu/microarchitectures.json:1519]: I verified that 2021.2 is the first compile to accept sapphirerapids

**Comment by stephenmsachs** [comment]
@alalazo can we merge or do we need need any more reviews?

**Comment by alalazo** [comment]
@stephenmsachs Can you add a sample of a `/proc/cpuinfo` from a bare metal sapphire rapids for the tests?

**Comment by stephenmsachs** [comment]
This is from a `m7i.24xlarge`:
```
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 143
model name      : Intel(R) Xeon(R) Platinum 8488C
stepping        : 8
microcode       : 0x2b000461
cpu MHz         : 3101.940
cache size      : 107520 KB
physical id     : 0
siblings        : 48
core id         : 0
cpu cores       : 48
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 31
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq monitor ssse3 fma cx16 pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx512_bf16 wbnoinvd ida arat avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq rdpid cldemote movdiri movdir64b md_clear serialize flush_l1d arch_capabilities
bugs            : spectre_v1 spectre_v2 spec_store_bypass swapgs
bogomips        : 4800.00
clflush size    : 64
cache_alignment : 64
address sizes   : 46 bits physical, 48 bits virtual
power management:
```

**Review by alalazo — APPROVED** [review]

**Pull request merged**